torch/_higher_order_ops/effects.py

```diff
@@ -0,0 +1,198 @@
+from enum import Enum
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+import torch.utils._pytree as pytree
+from torch._C import DispatchKey
+from torch._ops import HigherOrderOperator
+from torch._subclasses.fake_tensor import FakeTensorMode
+from torch.fx.experimental.proxy_tensor import (
+    disable_proxy_modes_tracing,
+    ProxyTorchDispatchMode,
+    track_tensor_tree,
+)
+
+
+SIDE_EFFECTFUL_OPS = {torch.ops.aten._print.default}
```

Comment:
I'm not sure how to distinguish the two. This list seems to be for ops such that if you reorder them, some correctness issue will occur, but the FX list seems to be to prevent certain ops with users=0 from being DCE-d. I changed this list to a map instead though, hopefully that could help?